src/mozanalysis/experiment.py

```diff
@@ -250,57 +265,63 @@ def get_time_series_data(
         """
 
         time_limits = TimeLimits.for_ts(
-            self.start_date, last_date_full_data, time_series_period,
-            self.num_dates_enrollment
+            self.start_date,
+            last_date_full_data,
+            time_series_period,
+            self.num_dates_enrollment,
         )
 
-        sql_template = self.build_query_template(
-            metric_list=metric_list,
+        sql_template_enrollments = self.build_enrollment_query(
             time_limits=time_limits,
             enrollments_query_type=enrollments_query_type,
             custom_enrollments_query=custom_enrollments_query,
             segment_list=segment_list,
         )
 
-        full_res_table_name = sanitize_table_name_for_bq('_'.join(
-            [last_date_full_data, self.experiment_slug, hash_ish(sql_template)]
-        ))
+        enrollments_table = bq_context.run(sql_template_enrollments)
 
-        sql = sql_template.format(results_table=full_res_table_name)
+        sql_template_metrics = self.build_metrics_query(
+            metric_list=metric_list,
+            time_limits=time_limits,
+            enrollments_table=enrollments_table,
+        )
+
+        full_res_table_name = sanitize_table_name_for_bq(
+            "_".join(
+                [
+                    last_date_full_data,
+                    self.experiment_slug,
+                    hash_ish(sql_template_metrics),
+                ]
+            )
+        )
 
-        bq_context.run_script_or_fetch(sql, full_res_table_name)
+        bq_context.run_query_and_fetch(sql_template_metrics, full_res_table_name)
```

Comment:
This is more or less what jetstream will do. The enrollments and metrics query can then be validated in a dry run